frontend/azlinux/handle_bin.go

```diff
@@ -0,0 +1,107 @@
+package azlinux
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend"
+	"github.com/Azure/dalec/frontend/rpm"
+	"github.com/moby/buildkit/client/llb"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+func binCopyScript(rpms []string, binaries map[string]dalec.ArtifactConfig) string {
+	var sb strings.Builder
+
+	sb.WriteString(`
+#!/bin/bash
+set -e
+declare -a RPMS=()
+export RPM_BINDIR=$(rpm --eval '%{_bindir}')
+`)
+
+	for _, rpm := range rpms {
+		sb.WriteString(fmt.Sprintf("RPMS+=(%q)\n", rpm))
+	}
```

Comment:
Bash has built-in ways of building up arrays from text. My personal choice is to use [readarray](https://helpmanual.io/builtin/readarray/) (an alias of `mapfile`). You have to be sure you're using bash and not some other shell, and you have to be sure that your bash version is >= 4 (almost all distros these days).

You could get the data into the container by
```go
rpmsData := []byte(strings.Join(rpms, "\x00")) // null separate the rpms
```

Then, you can take those bytes and use `llb.MkFile` to mount it into the container, for example at `/tmp/rpm_list`. In bash, you would do:

```bash
readarray -d '' RPMS < /tmp/rpm_list # reads the file and puts the results into the array variable "$RPMS"
```

This note is for educational purposes, what you have is fine.

Reply:
Thanks for this, I am not super familiar with bash array building capabilities, so I appreciate it!